src/lib.rs

```diff
@@ -898,7 +926,8 @@ extern "C" fn fn_encrypt_init(
     let mech = res_or_ret!(token.get_mechanisms().get(data.mechanism));
     if mech.info().flags & CKF_ENCRYPT == CKF_ENCRYPT {
         let operation = res_or_ret!(mech.encryption_new(data, &obj));
-        session.set_operation(Operation::Encryption(operation));
+        session
+            .set_operation(Operation::Encryption(operation), obj.always_auth());
```

Comment:
Does it make sense to check for this on public key operation which generally does not need any authentication?

Reply:
The attribute is available only on private key objects

Reply:
Oh, I get it, I just applied these to all operations and didn't think encryption and verification does not need it indeed ... yeah I will change those two places, good catch.